packages/dev/core/src/AudioV2/abstractAudio/abstractSound.ts

```diff
@@ -0,0 +1,308 @@
+    }
+
+    public set currentTime(value: number) {
+        this.startOffset = value;
+
+        const instance = this._getNewestInstance();
+        if (instance) {
+            instance.currentTime = value;
+        }
+    }
+
+    /**
+     * Whether the sound should loop. Defaults to `false`.
+     */
+    public get loop(): boolean {
+        return this._options.loop;
+    }
+
+    public set loop(value: boolean) {
+        this._options.loop = value;
+    }
+
+    /**
+     * The maximum number of instances that can play at the same time. Defaults to `Infinity`.
+     */
+    public get maxInstances(): number {
+        return this._options.maxInstances;
+    }
+
+    public set maxInstances(value: number) {
+        this._options.maxInstances = value;
+    }
+
+    /**
+     * The output bus for the sound. Defaults to the audio engine's default main bus.
+     * @see {@link AudioEngineV2.defaultMainBus}
+     */
+    public get outBus(): Nullable<PrimaryAudioBus> {
+        return this._outBus;
+    }
+
+    public set outBus(outBus: Nullable<PrimaryAudioBus>) {
+        if (this._outBus === outBus) {
+            return;
+        }
+
+        if (this._outBus) {
+            this._disconnect(this._outBus);
+        }
+
+        this._outBus = outBus;
+
+        if (this._outBus) {
+            this._connect(this._outBus);
+        }
+    }
+
+    /**
+     * The spatial properties of the sound.
+     */
+    public abstract get spatial(): AbstractSpatialAudio;
+
+    /**
+     * The time within the sound buffer to start playing at, in seconds. Defaults to `0`.
+     */
+    public get startOffset(): number {
+        return this._options.startOffset;
+    }
+
+    public set startOffset(value: number) {
+        this._options.startOffset = value;
+    }
+
+    /**
+     * The state of the sound.
+     */
+    public get state(): SoundState {
+        return this._state;
+    }
+
+    /**
+     * The stereo properties of the sound.
+     */
+    public abstract get stereo(): AbstractStereoAudio;
+
+    /**
+     * The output volume of the sound.
+     */
+    public get volume(): number {
+        return this._subGraph.getSubNode<_VolumeAudioSubNode>(_AudioSubNode.Volume)?.volume ?? _VolumeAudioDefaults.Volume;
+    }
+
+    public set volume(value: number) {
+        // Note that the volume subnode is created at initialization time and it always exists, so the callback that
+        // sets the node's volume is always called synchronously.
+        this._subGraph.callOnSubNode<_VolumeAudioSubNode>(_AudioSubNode.Volume, (node) => {
+            node.volume = value;
+        });
+    }
+
+    protected get _isPaused(): boolean {
+        return this._state === SoundState.Paused && this._instances.size > 0;
+    }
+
+    /**
+     * Releases associated resources.
+     */
+    public override dispose(): void {
+        super.dispose();
+
+        this.stop();
+
+        this._outBus = null;
```

Comment:
no need to dispose it? Or is it disposed somewhere else and is only referenced here?

Reply:
Ya, multiple sounds can reference a bus.

It's not apparent from only looking at the abstract classes, but the audio engine owns all top-level nodes like sounds and buses, and the audio engine functions to add and remove the top-level nodes are only exposed publicly in subclasses (`WebAudioEngine` in this case). Subclasses for top-level nodes need to add themselves to the audio engine, which is done in the `WebAudio` subclasses' init functions.

Looking at this more, I'm realizing a sound can be referencing a bus that's been disposed. I'll handle that case.